### ISSUE TITLE
[Bug]: shopify theme push isn't working on 3.59.0 again

### ISSUE THREAD

**Issue by xmadscientist** [Shopify/cli#3731]
### Please confirm that you have:

- [X] Searched [existing issues](.) to see if your issue is a duplicate. (If you’ve found a duplicate issue, feel free to add additional information in a comment on it.)
- [X] Reproduced the issue in [the latest CLI version](https://www.npmjs.com/package/@shopify/cli).

### In which of these areas are you experiencing a problem?

Theme

### Expected behavior

Push changed files.

### Actual behavior

Not pushing changed files.

### Verbose output

```shell
https://pastebin.com/Yq4L6Kjz

(output was too long to open a ticket)
```

### Reproduction steps

1. `shopify theme push -e=env_name --allow-live`

### Operating System

Mac OS Ventura

### Shopify CLI version (check your project's `package.json` if you're not sure)

3.59.0

### Shell

zsh

### Node version (run `node -v` if you're not sure)

v14.18.3

### What language and version are you using in your application?

N/A

**Comment by mgmanzella** [comment]
👋 hi @xmadscientist thank you for reporting this issue, would you mind rerunning the command you shared but with `--verbose`? we need this information in order to debug:

`shopify theme push -e=env_name --allow-live --verbose`

thank you for your help 🙏

**Comment by stijns96** [comment]
Hi @mgmanzella,

When can we expect a fix? For our agency it's quite crucial to get the ignore file on point. Otherwise we are overwriting the content of our clients... Previously we used `/(locales\/)(?!en-dev.default.json)/`, but this doesn't seem to work anymore as well.

**Comment by mgmanzella** [comment]
i don't have a time estimate on when this will be fixed, we're working through edge cases of glob matching being different in the TS implementation of `theme push` vs Ruby. 

try using the `--stable` flag with your command, it should allow it to work as expected but let us know if it doesn't 🙏

**Comment by CodeBrotha** [comment]
Same problem here. This issue surfaced previously in 3.58.0 and was fixed in 3.58.2.

It seems that fix was reverted for some reason.

**Comment by jamesmengo** [comment]
@xmadscientist Closing this issue since there is a workaround with the `--stable` flag

Feel free to re-open with more information and the full verbose log if needed!


For issues relating to your `shopifyignore`, please add to
- #3749 

Thanks :)
